cpp/src/arrow/util/thread_pool_benchmark.cc

```diff
@@ -103,8 +103,40 @@ static void ThreadPoolSpawn(benchmark::State& state) {
   state.SetItemsProcessed(state.iterations() * nspawns);
 }
 
+// Benchmark ThreadPool::Submit
+static void ThreadPoolSubmit(benchmark::State& state) {  // NOLINT non-const reference
+  const auto nthreads = static_cast<int>(state.range(0));
+  const auto workload_size = static_cast<int32_t>(state.range(1));
+
+  Workload workload(workload_size);
+
+  const int32_t nspawns = 10000000 / workload_size + 1;
+
+  for (auto _ : state) {
+    state.PauseTiming();
+    auto pool = *ThreadPool::Make(nthreads);
+    std::atomic<int32_t> n_finished{0};
+    state.ResumeTiming();
+
+    for (int32_t i = 0; i < nspawns; ++i) {
+      // Pass the task by reference to avoid copying it around
+      (void)DeferNotOk(pool->Submit(std::ref(workload))).Then([&](...) {
```

Comment:
(the `Future` micro-benchmarks can be deferred to another PR, though)

Reply:
I agree it warrants improvement and individual benchmarking, but my intent was to measure the end-to-end cost of using Submit+callbacks since that's relevant to the csv parsing case